File: 345-insertNodeInLinkList.cpp
#include"linkNode.h"

#include<iostream>


LinkNode* insertNodeByPos(LinkNode* head, int val, int pos)
{
    if(!head)
        return new LinkNode(val);
    LinkNode *dummyHead=new LinkNode(0);
    dummyHead->next=head;
    LinkNode *preNode=dummyHead;
    for(int i=0;i<pos;++i)
    {
        preNode=head;
        head=head->next;
    }
    LinkNode* newNode=new LinkNode(val);
    preNode->next=newNode;
    newNode->next=head;

    return dummyHead->next;
}


int main()
{
    LinkNode* firstNode=new LinkNode(1);
    LinkNode* secNode=new LinkNode(3);
    firstNode->next=secNode;

    firstNode->printNode();
    LinkNode* newLinkList=insertNodeByPos(firstNode,2,1);
    newLinkList->printNode();
    newLinkList=insertNodeByPos(firstNode,4,3);
    newLinkList->printNode();

    return 0;
}

File: 345-insertNodeInLinkList.py
class Node:
    def __init__(self,val=None,next=None):
        self.val=val
        self.next=next
        
        


def insert(head,val,pos):
    if not head:
        return Node(val)
    newNode=Node(val)
    dummy=Node(0,head)
    preNode=dummy
    
    for _ in range(pos):
        preNode=head
        head=head.next

    preNode.next=newNode
    newNode.next=head
    return dummy.next

fNode=Node(1)
sNode=Node(2)

